second wayland issue

from the output of "make" 

up to date opensuse tumbleweed 
KDE desktop using Wayland 


--------- quote --------
`[ 77%] Building CXX object src/celestia/qt6/CMakeFiles/celestia-qt6.dir/__/qt/qtdraghandler.cpp.o
In file included from /DATA/SUSE/src/Celestia/src/celestia/qt/qtwaylanddraghandler.h:8,
                 from /DATA/SUSE/src/Celestia/src/celestia/qt/qtdraghandler.cpp:11:
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h: In function ‘void zwp_pointer_constraints_v1_set_user_data(zwp_pointer_constraints_v1*, void*)’:
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h:297:9: error: ‘wl_proxy_set_user_data’ was not declared in this scope
  297 |         wl_proxy_set_user_data((struct wl_proxy *) zwp_pointer_constraints_v1, user_data);
      |         ^~~~~~~~~~~~~~~~~~~~~~
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h: In function ‘void* zwp_pointer_constraints_v1_get_user_data(zwp_pointer_constraints_v1*)’:
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h:304:16: error: ‘wl_proxy_get_user_data’ was not declared in this scope
  304 |         return wl_proxy_get_user_data((struct wl_proxy *) zwp_pointer_constraints_v1);
      |                ^~~~~~~~~~~~~~~~~~~~~~
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h: In function ‘uint32_t zwp_pointer_constraints_v1_get_version(zwp_pointer_constraints_v1*)’:
/DATA/SUSE/src/Celestia/build/pointer-constraints-unstable-v1-client-protocol.h:310:16: error: ‘wl_proxy_get_version’ was not declared in this scope; did you mean ‘epoxy_gl_version’?
  310 |         return wl_proxy_get_version((struct wl_proxy *) zwp_pointer_constraints_v1);
      |                ^~~~~~~~~~~~~~~~~~~~
      |                epoxy_gl_version

    --- edited out - This goes on for a bit -- long error  -----

make[2]: *** [src/celestia/qt6/CMakeFiles/celestia-qt6.dir/build.make:251: src/celestia/qt6/CMakeFiles/celestia-qt6.dir/__/qt/qtdraghandler.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2408: src/celestia/qt6/CMakeFiles/celestia-qt6.dir/all] Error 2
make: *** [Makefile:136: all] Error 2



`


--------------------- end quote -----------

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,7 +318,7 @@ if(USE_WAYLAND)
 
   set_source_files_properties(${PROTOCOL_SOURCES} PROPERTIES GENERATED ON)
   add_library(wayland-protocols-helper STATIC ${PROTOCOL_SOURCES})
-  target_include_directories(wayland-protocols-helper INTERFACE ${CMAKE_CURRENT_BINARY_DIR})
+  target_include_directories(wayland-protocols-helper PUBLIC ${Wayland_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 endif()
 
 find_package(gperf REQUIRED)

